app/controllers/check_controller.rb

```diff
@@ -0,0 +1,20 @@
+class CheckController < ApplicationController
+  def new
+    @checks = Check.new
+  end
+
+  def show
+    @checks = Check.find(params[:id])
```

Comment:
You are looking for a Check in the database that is not saved in the database, by an ID that is not being submitted.  Remove this line and things should work better.